src/vivarium_public_health/risks/effect.py

```diff
@@ -316,3 +320,164 @@ def _get_risk_exposure_class(self, builder: Builder) -> Risk:
                 f"Risk effect model {self.name} requires a Risk component named {self.risk}"
             )
         return risk_exposure_component
+
+
+class NonLogLinearRiskEffect(RiskEffect):
+    """Risk effect for exposure-parametrized relative risks."""
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def configuration_defaults(self) -> Dict[str, Any]:
+        """
+        A dictionary containing the defaults for any configurations managed by
+        this component.
+        """
+        return {
+            self.name: {
+                "data_sources": {
+                    "relative_risk": f"{self.risk}.relative_risk",
+                    "population_attributable_fraction": f"{self.risk}.population_attributable_fraction",
+                },
+            }
+        }
+
+    @property
+    def columns_created(self) -> List[str]:
+        return [f"{self.risk.name}_exposure"]
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        exposure_col = pd.DataFrame(columns=self.columns_created)
+        self.population_view.update(exposure_col)
+
+    def on_time_step_prepare(self, event: Event) -> None:
+        # exposure_values = self.exposure(event.index)
+        # TODO: temporary workaround, investigate why exposure is always 0
+        exposure_values = np.linspace(3, 7, num=len(event.index))
+        exposure_col = pd.DataFrame(exposure_values, columns=self.columns_created).astype(
+            "object"
+        )
+        self.population_view.update(exposure_col)
+
+    #################
+    # Setup methods #
+    #################
+
+    def build_all_lookup_tables(self, builder: Builder) -> None:
+        rr_data = self.get_relative_risk_data(builder)
+
+        # TODO: add check that rr_data is parametrize by exposure
+        def define_rr_intervals(df: pd.DataFrame) -> pd.DataFrame:
+            max_exposure_row = df.tail(1).copy()
+            max_exposure_row["parameter"] = 9999
+            rr_data = pd.concat([df, max_exposure_row]).reset_index()
+            rr_data["left_exposure"] = [0] + rr_data["parameter"][:-1].tolist()
+            rr_data["left_rr"] = [rr_data["value"].min()] + rr_data["value"][:-1].tolist()
```

Comment:
Is this forcing the left_rr.min() at exposure 0? Is that the correct thing to do?

Reply:
Good question - we do want to 1) fill all RR values below the lowest exposure to be the smallest relative risk which 2) will be 1 anyway because this exposure should be the TMREL.